lib/Fmt_ast.ml

```diff
@@ -4126,8 +4126,8 @@ and fmt_structure_item c ~last:last_item ?ext {ctx; ast= si} =
         || (not (List.is_empty itmJ.pvb_attributes))
         || Ast.break_between c.source ~cmts:c.cmts
              ~has_cmts_before:Cmts.has_before ~has_cmts_after:Cmts.has_after
-             (Exp itmI.pvb_expr, cI.conf)
-             (Exp itmJ.pvb_expr, cJ.conf)
+             (Exp {itmI.pvb_expr with pexp_loc= itmI.pvb_loc}, cI.conf)
+             (Exp {itmJ.pvb_expr with pexp_loc= itmJ.pvb_loc}, cJ.conf)
```

Comment:
I didn't see a case for value_binding in place of using `Exp` here, but that would be a less-hacky fix. I'm not sure how interconnected with other things adding such a constructor would be.

Reply:
I think it would be better to add a constructor in the `Ast.t` type for each kind of ast element, I don't think the constructors of this type are interconnected.

Reply:
Done.